### PR TITLE
Typography updates

### PR DESCRIPTION
**Pull request by bobwise**
## What does this PR do?
* Updates Typography design tokens for all typography types.
* New content and design for documentation page.

Preview Page: https://deploy-preview-4045--spark-design-system.netlify.app/using-spark/foundations/typography

### Associated Issue
Fixes 2531140

### Documentation
 - [x] Update Spark Docs
 
### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)
  
### Accessibility Testing
  - [x] Axe browser extension
  - [ ] Jaws Inspect (License expired)
  - [x] VoiceOver (iOS) or Talkback (Android)

### Deploy Preview Reviewed and Approved
 - [x] Design Team (In progress)
 - [x] Accessibility Team (In progress)

### CODE REVIEW

File: src/pages/using-spark/foundations/typography.mdx [hunk 1]
```diff
@@ -95,15 +136,191 @@ Our guidelines optimize content hierarchy by using text sizes, line heights, col
...
 ## Headers
 
 ### Page Title
 
 Use one Page Title per page to serve as the main heading. They differentiate themselves from other headings with their red accent line.
 
-Use either [Display One](#display-one) or [Display Two](display-two) to style this heading.
+Use [Display One](#display-one) to style level 1, or top-level pages in your experience.
+
+Use [Display Two](display-two) to style level 2, an inner content page, or below pages in your experience.
```
Comment (guern1kn) [inline src/pages/using-spark/foundations/typography.mdx:323]: I think you fixed this in `main`? The (display-two) link needs to be fixed.

File: styles/base/inputs/_svg-icon-input.scss [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
 .sprk-b-InputContainer__icon-container .sprk-b-InputContainer__icon,
 .sprk-b-TextInputIconContainer .sprk-b-InputContainer__icon,
+.sprk-b-TextInputIconContainer .sprk-c-Icon,
```
Comment (guern1kn) [inline styles/base/inputs/_svg-icon-input.scss:16]: WAsn't this one of the things that was causing an issue? Didn't @RVMendoza need to remove this is favor of what's on line 15?

Comment (bobwise) [reply styles/base/inputs/_svg-icon-input.scss:16]: Hmm I added this back to fix a problem with date picker. I'll touch base with RV

File: src/pages/using-spark/foundations/typography.mdx [hunk 3]
```diff
@@ -95,15 +136,191 @@ Our guidelines optimize content hierarchy by using text sizes, line heights, col
...
+<SprkDivider element="span" additionalClasses="sprk-u-mvn" />
+
+## Size Reference Charts
+
+Our typography has 7 display styles, 3 body copy styles, and 1 legal type size.
+The size reference chart is an example of typography when designing.
+However, when developing a page, reference the Storybook components below.
+
+<div class="sprk-o-Stack sprk-o-Stack--medium sprk-o-Stack--split@xl">
+  <div class="sprk-b-TableContainer sprk-o-Stack__item sprk-u-mtl">
```
Comment (afebbraro) [inline src/pages/using-spark/foundations/typography.mdx:148]: Do we need to use a the !important util here?

File: src/pages/using-spark/foundations/typography.mdx [hunk 4]
```diff
@@ -95,15 +136,191 @@ Our guidelines optimize content hierarchy by using text sizes, line heights, col
...
+          <td>Legal</td>
+          <td>Light</td>
+          <td>13px</td>
+          <td>20px</td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+
+  <div class="sprk-b-TableContainer sprk-o-Stack__item sprk-u-mtl">
```
Comment (afebbraro) [inline src/pages/using-spark/foundations/typography.mdx:230]: Do we need to use a the !important util here?

File: src/pages/using-spark/foundations/typography.mdx [hunk 5]
```diff
@@ -113,12 +330,107 @@ Use either [Display One](#display-one) or [Display Two](#display-two) to style t
...
 />
+
 <div class="sprk-u-Measure">
   <a class="sprk-b-Link sprk-u-Float--right" href="#table-of-contents">
     Back to Table of Contents
   </a>
 </div>
 
+<SprkFlag
+  additionalClasses="sprk-o-Flag--middle sprk-u-mtl sprk-u-pvm"
```
Comment (afebbraro) [inline src/pages/using-spark/foundations/typography.mdx:341]: Do we need to use a the !important util here?

**Review by afebbraro — CHANGES_REQUESTED** [review]
Left a few comments on the code. Checking out the deploy previews now.

File: styles/base/inputs/_svg-icon-input.scss (hunk 2 above)
Comment (bobwise) [reply styles/base/inputs/_svg-icon-input.scss]: Update: I talked with RV and ended up rolling this change back. I talked to Steph about making this a separate Issue.

File: styles/base/_typography.scss [hunk 6]
```diff
@@ -180,6 +222,20 @@ h6,
...
+
+  @media screen and (min-width: $sprk-type-display-seven-breakpoint) {
+    /* prettier-ignore */
+
+    @include TextCrop(
+      $sprk-text-crop-line-height: $sprk-line-height-display-seven-wide,
+      $sprk-text-crop-top-crop: $sprk-line-height-collision-top-crop,
+      $sprk-text-crop-bottom-crop: $sprk-line-height-collision-bottom-crop,
+      $sprk-text-crop-crop-font-size: (strip-unit($sprk-font-size-display-seven-wide) * 16),
+      $sprk-text-crop-crop-line-height: $sprk-line-height-display-seven
```
[the remark this answers is not in the document]
Comment (bobwise) [reply styles/base/_typography.scss:234]: Good catch!

**Review by RVMendoza — APPROVED** [review]
image
spacing weirdness here, but other than that + Amber's suggestions, this looks good!

**Comment by bobwise** [comment]
All feedback has been addressed. We may still update the "specimen" image at the top of the page.

**Review by afebbraro — APPROVED** [review]

**Pull request merged**